deepfence_server/model/registry.go

```diff
@@ -389,28 +390,116 @@ func ListImages(ctx context.Context, registryId string, filter reporters.FieldsF
 		return res, err
 	}
 
+	/*
+		Logic to prepare the image list for a repository:
+		Step1: Prepare a map "stubTags" of all the valid tags for this repository.
+		For this, we take the "tags" from the ImageStub and use that to populate
+		the map. Also get the name of the image from the ImageStub as "dockerImageName"
+
+		Step2: Iterate through the matched ContainerImage records.
+		If the name of image on this ContainerImage match dockerImageName:
+			Iterate through the tags for this ContainerImage and only keep one's
+			which are in the "stubTags". After adding, delete the tag from the stubTags.
+			Add this ContainerImage to the final list
+		If the name of the image on this ContainerImage does not match dockerImageName:
+			Add this to the exclusion list "nodeListEx"
+
+		Step3: Iterate the ContainerImage nodes in nodeListEx.
+		Iterate through the tags for this ContainerImage and only keep one's
+		which are in the "stubTags". After adding, delete the tag from the stubTags.
+		Add this ContainerImage to the final list
+	*/
+
+	//Step1: Prepare a map "stubTags" of all the valid tags for this repository.
+	stubTags := make(map[string]bool)
```

Comment:
Hum.. I fail to see how this will work, `subTags` will end up with tags from a set of images that are not related to each others. We could end up in situation like:
```
Repo A: Image A: Tag A
Repo A: Image B: Tag B
Repo B: Image A: Tag B
```
Now you will end up with `Repo A: Image A: Tag A` which does not exist

Reply:
The `stubTags` are set of tags from a single Repository. So for the above example, for Repo A, we will have `stubTags` as {Tag A, Tag B}. And for Repo B, `stubTags` would be {Tag B}.